test/integration/shoots/operations/clusterautoscaler.go

```diff
@@ -165,6 +166,125 @@ var _ = ginkgo.Describe("Shoot clusterautoscaler testing", func() {
 		framework.ExpectNoError(err)
 	}, testTimeout)
 
+	f.Beta().Serial().CIt("should autoscale a single worker group to/from zero", func(ctx context.Context) {
+		var (
+			shoot = f.Shoot
+
+			origClusterAutoscalerConfig = shoot.Spec.Kubernetes.ClusterAutoscaler.DeepCopy()
+			workerPoolName              = shoot.Spec.Provider.Workers[0].Name
+			testWorkerPoolName          = "ca-test"
+			origWorkers                 = shoot.Spec.Provider.Workers
+		)
+		if shoot.Spec.Provider.Type != "aws" && shoot.Spec.Provider.Type != "azure" {
+			ginkgo.Skip("not applicable")
+		}
+
+		// Create a dedicated worker-pool for cluster autoscaler.
+		testWorkerPool := shoot.Spec.Provider.Workers[0]
+		testWorkerPool.Name = testWorkerPoolName
+		testWorkerPool.Minimum = 0
+		testWorkerPool.Maximum = 2
+		testWorkerPool.Labels = map[string]string{"worker.gardener.cloud/pool": testWorkerPoolName}
+		testTaint := corev1.Taint{
+			Key:    testWorkerPoolName,
+			Effect: corev1.TaintEffectNoSchedule,
+			Value:  testWorkerPoolName,
+		}
+
+		testWorkerPool.Taints = append(testWorkerPool.Taints, testTaint)
+
+		ginkgo.By("updating shoot spec for test")
+		err := f.UpdateShoot(ctx, func(s *corev1beta1.Shoot) error {
+			s.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, testWorkerPool)
+
+			if s.Spec.Kubernetes.ClusterAutoscaler == nil {
+				s.Spec.Kubernetes.ClusterAutoscaler = &corev1beta1.ClusterAutoscaler{}
+			}
+			s.Spec.Kubernetes.ClusterAutoscaler.ScaleDownDelayAfterAdd = &metav1.Duration{Duration: scaleDownDelayAfterAdd}
+			s.Spec.Kubernetes.ClusterAutoscaler.ScaleDownUnneededTime = &metav1.Duration{Duration: scaleDownUnneededTime}
+
+			return nil
+		})
+		framework.ExpectNoError(err)
+
+		// Reset shoot spec to original values after test
+		defer func() {
+			ginkgo.By("reverting shoot spec changes by test")
+			err := f.UpdateShoot(ctx, func(s *corev1beta1.Shoot) error {
+				s.Spec.Kubernetes.ClusterAutoscaler = origClusterAutoscalerConfig
+
+				// Remove the dedicated ca-test workerpool
+				s.Spec.Provider.Workers = origWorkers
```

Comment:
The test is Serial() thats why this is feasible I guess. 
What if you change the logic to only remove the added worker pool instead of completely overwriting it?  This way the test may be able  to work in non-serial mode? (pool also has taints so that no other pod is scheduled there).
Or could the changed autoscaler config create problems in parallely executed tests?

Reply:
>Or could the changed autoscaler config create problems in parallely executed tests?

It should not really be an issue running it in parallel in the future.  I'll make a change to remove the exact worker pool than overwriting to the original one, good point. Thanks.

Reply:
done.